polars/polars-sql/src/functions.rs

```diff
@@ -366,6 +374,20 @@ impl SqlFunctionVisitor<'_> {
             Log1p => self.visit_unary(Expr::log1p),
             Log2 => self.visit_unary(|e| e.log(2.0)),
             Pow => self.visit_binary::<Expr>(Expr::pow),
+            Round => match function.args.len() {
+                1 => self.visit_unary(|e| e.round(0)),
+                2 => self.try_visit_binary(|e, decimals| {
```

Comment:
It was a bit more complex, because we were in `visit_binary` that accepts a `|args| -> Expr` and we needed a `try_visit_binary` that accepts a `|args| -> Result<Expr>`.

Reply:
Nice; we should probably implement this pattern in `visit_function` too (bail rather than panic) 🤔

Reply:
Yes, we should.